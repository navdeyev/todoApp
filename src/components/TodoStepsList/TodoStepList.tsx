import * as React from 'react';

import {ITodoStep} from 'domains/todos/todosTypes';

import styled from 'components/Styled/styledComponents';
import {rem} from 'components/Styled/utils';

export const OrderedList = styled.ol`margin: 0;`;
OrderedList.displayName = 'OrderedList';

const ListItem = styled.li`
    padding-bottom: ${ rem(10) };
    
    &:first-child {
      padding-top: ${ rem(10) };
    }
    
    &:last-child {
      padding-bottom: 0;  
    }
`;
ListItem.displayName = 'ListItem';

const Bold = styled.span`
    font-weight: bold;
    margin-right: ${ rem(5) };
`;
Bold.displayName = 'Bold';

export interface IProps {
    steps: ITodoStep[],
    dataRole: string
}

const TodoStepList: React.SFC<IProps> = (props) => {
    const {steps, dataRole} = props;
    return (
        <OrderedList data-role={ dataRole }>
            {steps.map((step: ITodoStep) => {
                return <ListItem key={step.id}>
                    <Bold>{step.title}:</Bold>
                    <span>{step.details}</span>
                </ListItem>
            })}
        </OrderedList>
    );
};

export default TodoStepList;
