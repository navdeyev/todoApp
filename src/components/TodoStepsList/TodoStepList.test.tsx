import {shallow} from 'enzyme';
import 'jest-styled-components';
import * as React from 'react';

import TodoStepList, {IProps} from './TodoStepList';

describe('TodoStepList', () => {

    let defaultProps: IProps;
    beforeEach(() => {
        defaultProps = {
            steps: [{
                details: 'Step details',
                id: 'todo-3-step-1',
                title: 'Step tile'
            }],
            dataRole: 'todo-card-step-list-0'
        }
    });

    it('renders a component', () => {
        const render = shallow(<TodoStepList {...defaultProps} />);
        expect(render.getElement()).toMatchSnapshot();
    });

});
