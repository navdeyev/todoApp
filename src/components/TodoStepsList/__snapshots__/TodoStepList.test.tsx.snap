// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoStepList renders a component 1`] = `
<OrderedList
  data-role="todo-card-step-list-0"
>
  <ListItem>
    <Bold>
      Step tile
      :
    </Bold>
    <span>
      Step details
    </span>
  </ListItem>
</OrderedList>
`;
